kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/parquet/ParquetColumnReaderSuite.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.internal.parquet
+
+import java.util.TimeZone
+
+import io.delta.golden.GoldenTableUtils.goldenTablePath
+import io.delta.kernel.defaults.utils.TestRow
+import io.delta.kernel.types._
+import org.scalatest.funsuite.AnyFunSuite
+
+/**
+ * Suite covering reading Parquet columns with different types.
+ */
+class ParquetColumnReaderSuite  extends AnyFunSuite with ParquetSuiteBase {
+
+  /**
+   * Defines a test case for this suite.
+   * @param columnName Column to read from the file
+   * @param toType Read type to use. May be different from the actualy Parquet type.
+   * @param expectedExpr Expression returning the expected value for each row in the file.
+   */
+  case class TestCase(columnName: String, toType: DataType, expectedExpr: Int => Any)
+
+  private val wideningTestCases: Seq[TestCase] = Seq(
+    TestCase("ByteType", ShortType.SHORT, i => if (i % 72 != 0) i.toByte.toShort else null),
+    TestCase("ByteType", IntegerType.INTEGER, i => if (i % 72 != 0) i.toByte.toInt else null),
+    TestCase("ByteType", LongType.LONG, i => if (i % 72 != 0) i.toByte.toLong else null),
```

Comment:
is byte to float or double not allowed?

Reply:
Byte to double is allowed, byte to float isn't. I added more test cases